crates/heat-sdk-cli/src/cli_commands/run/training.rs

```diff
@@ -0,0 +1,194 @@
+use clap::Parser;
+use colored::Colorize;
+use heat_sdk::{
+    client::{HeatClient, HeatClientConfig, HeatCredentials},
+    schemas::ProjectPath,
+};
+
+use crate::registry::Flag;
+use crate::{
+    commands::{execute_sequentially, BuildCommand, RunCommand, RunParams},
+    context::HeatCliContext,
+    generation::backend::BackendType,
+    logging::BURN_ORANGE,
+    print_info,
+};
+
+#[derive(Parser, Debug)]
+pub struct TrainingRunArgs {
+    /// The training functions to run
+    #[clap(short = 'f', long="functions", value_delimiter = ' ', num_args = 1.., required = true, help = "The training functions to run. Annotate a training function with #[heat(training)] to register it."
+    )]
+    functions: Vec<String>,
+    /// Backend to use
+    #[clap(short = 'b', long = "backends", value_delimiter = ' ', num_args = 1.., required = true, help = "Backends to use for training."
+    )]
+    backends: Vec<BackendType>,
+    /// Config files paths
+    #[clap(short = 'c', long = "configs", value_delimiter = ' ', num_args = 1.., required = true, help = "Config files paths."
+    )]
+    configs: Vec<String>,
+    /// The Heat project path
+    // todo: support project name and creating a project if it doesn't exist
+    #[clap(
+        short = 'p',
+        long = "project",
+        required = true,
+        help = "The Heat project path."
+    )]
+    project_path: String,
+    /// The Heat API key
+    #[clap(short = 'k', long = "key", required = true, help = "The Heat API key.")]
+    key: String,
+    /// Project version
+    #[clap(short = 't', long = "version", help = "The project version.")]
+    project_version: Option<String>,
+    /// The runner group name
+    #[clap(short = 'r', long = "runner", help = "The runner group name.")]
+    runner: Option<String>,
+}
+
+pub(crate) fn handle_command(args: TrainingRunArgs, context: HeatCliContext) -> anyhow::Result<()> {
+    match (&args.runner, &args.project_version) {
+        (Some(_), Some(_)) => remote_run(args, context),
+        (None, None) => local_run(args, context),
+        (Some(_), None) => Err(anyhow::anyhow!("You must provide the project version to run on the runner with --version argument")),
+        (None, Some(_)) => Err(anyhow::anyhow!("Project version is ignored when executing locally (i.e. no runner is defined with --runner argument"))
```

Comment:
I think we can make it just a warning and continue executing locally. 
The message can be tweaked like that:
"Project version is ignored when executing locally (i.e. no runner is defined with --runner argument). Running locally uses the current version on the local machine."